### PR TITLE
Rework handling of qubit mappings in `BraketBackend`

### LINKED ISSUE #7: `process_circuits` fails with `ValidationException` on density-matrix backend
Code:

```python
from pytket.circuit import Circuit
from pytket.extensions.braket import BraketBackend

b = BraketBackend(device="dm1", device_type="quantum-simulator", provider="amazon")
c = Circuit(1).Rz(0.5, 0)
b.process_circuit(c)
```

Output:
```
Traceback (most recent call last):
  File "dmbug.py", line 8, in <module>
    b.process_circuit(c)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/pytket/backends/backend.py", line 248, in process_circuit
    return self.process_circuits(
  File "/home/alec/r/pytket-braket/pytket/extensions/braket/backends/braket.py", line 708, in process_circuits
    task = self._run(bkcirc, n_shots=n_shots)
  File "/home/alec/r/pytket-braket/pytket/extensions/braket/backends/braket.py", line 639, in _run
    return self._device.run(bkcirc, self._s3_dest, shots=n_shots, **kwargs)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/braket/aws/aws_device.py", line 147, in run
    return AwsQuantumTask.create(
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/braket/aws/aws_quantum_task.py", line 156, in create
    return _create_internal(
  File "/usr/lib/python3.8/functools.py", line 875, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/braket/aws/aws_quantum_task.py", line 532, in _
    task_arn = aws_session.create_quantum_task(**create_task_kwargs)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/braket/aws/aws_session.py", line 190, in create_quantum_task
    response = self.braket_client.create_quantum_task(**boto3_kwargs)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/botocore/client.py", line 508, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/alec/r/pytket-braket/env/lib/python3.8/site-packages/botocore/client.py", line 915, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the CreateQuantumTask operation: Unexpected token Token('$END', '') at line 5, column 41.
Expected one of: 
	* EQUALS
```

### LINKED ISSUE #9: `ValidationException` running small circuit on Aspen M-2
```python
from pytket.circuit import Circuit
from pytket.extensions.braket import BraketBackend

b = BraketBackend(
    device="Aspen-M-2", region="us-west-1", device_type="qpu", provider="rigetti"
)

c = Circuit(2)
c.H(0).CX(0, 1)
c.measure_all()

c1 = b.get_compiled_circuit(c)
h = b.process_circuit(c1, n_shots=100)
```

results in:
```
botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the CreateQuantumTask operation: Used 148 qubits, which is more than the device qubit count of 80
```

### LINKED ISSUE #10: [CI] Windows tests not running
The `build-test` script is executed with powershell instead of bash; it does nothing. We need to add `shell: bash` to this step in the workflow.

### PR DESCRIPTION
**Pull request by cqc-alec**
The logic was rather broken, in particular leading to errors when the device has qubits with non-contiguous indices (like the Aspen-M-2).

Update Rigetti tests to use Aspen-M-2, which seems to be the only available device now.

Ensure tests get run on Windows by running the script in the bash shell.

Ensure we get the fix to https://github.com/aws/amazon-braket-sdk-python/issues/430 by updating the amazon-braket-sdk requirement.

Update other dependencies too.

Fixes #7 
Fixes #9 
Fixes #10

### CODE REVIEW

**Comment by yao-cqc** [comment]
I pushed a commit to better handle the s3 destination arguments. Feel free to change it.

**Comment by cqc-alec** [comment]
> I pushed a commit to better handle the s3 destination arguments. Feel free to change it.

Thanks.

Not sure why the checks are not running. I will close and reopen the PR to see if that triggers them.

File: pytket/extensions/braket/backends/braket.py [hunk 1]
```diff
@@ -391,7 +416,10 @@ def __init__(
         )
 
         paradigm = props["paradigm"]
-        n_qubits = paradigm["qubitCount"]
+        n_qubits = len(self._all_qubits)
+
+        # TODO Is there a programmatic way to discover this (except trial and error)?
+        self._supports_client_qubit_mapping = provider == "rigetti"
```
Comment (yao-cqc) [inline pytket/extensions/braket/backends/braket.py:422]: `device_info["disabledQubitRewiringSupported"]` might be it. Here are the values for all available gate based devices.

Lucy: False
ionQdevice: False
Aspen-11: True
Aspen-M-2: True
dm1: None
tn1: None
sv1: None
braket_sv: None
braket_dm: None

**Review by yao-cqc — COMMENTED** [review]
Looks good. Just one suggestion for programmatically setting the manual qubit mapping flag.

**Review by yao-cqc — APPROVED** [review]

**Pull request merged**